main.py

```diff
@@ -0,0 +1,63 @@
+import pygame
+import serial
+import sympy
+import numpy
+import matplotlib
+import PySimpleGUI as sg
+from threading import Thread
+import time
+from camera import Camera
+from interface import Interface
+from controls import Control
+
+
+def gstreamer_pipeline(sensor_id = 0,sensor_mode = 3,capture_width = 1280,capture_height = 1280,display_width = 1280,display_height = 1280,framerate = 60,flip_method = 0):
+    return("nvarguscamerasrc sensor-id=%d sensor-mode=%d ! "
+    "video/x-raw(memory:NVMM), "
+    "width=(int)%d, height=(int)%d, "
+    "format=(string)NV12, framerate=(fraction)%d/1 ! "
+    "nvvidconv flip-method=%d ! "
+    "video/x-raw, width=(int)%d, height=(int)%d, format=(string)BGRx ! "
+    "videoconvert ! "
+    "video/x-raw, format=(string)BGR ! appsink"
+    % (
+        sensor_id,
+        sensor_mode,
+        capture_width,
+        capture_height,
+        framerate,
+        flip_method,
+        display_width,
+        display_height,
+    ))
+
+if __name__ == '__main__':
+    camera = Camera()
+    camera.open(gstreamer_pipeline(sensor_id=0, sensor_mode=3, flip_method=0, display_height=600, display_width=600))
+    camera.start()
+    controller = Control(100,0.1)
+    interface = Interface(robot=controller.list_movements, object_seen=True, motor_controls=True, clock=controller.clock)
```

Comment:
If the controller isn't connected, passing `controller.clock` will fail because `clock` was never instantiated. Check the `Control` class.